script.js

```diff
@@ -1,12 +1,169 @@
+
+    //Assigning TV episodes in p element
+    let episodeName = document.createElement("p");
+    episodeName.classList.add("epName");
+    movieCard.appendChild(episodeName);
 
+    //assigning the season number & episode number//
+    episodeSeason = episodeList[i].season;
+    episodeNumber = episodeList[i].number;
+    if (episodeSeason < 10) {
+      episodeSeason = "0" + episodeSeason;
+    }
+    if (episodeNumber < 10) {
+      episodeNumber = "0" + episodeNumber;
+    }
+    episodeName.innerText =
+      episodeList[i].name +
+      " " +
+      "-" +
+      " " +
+      "S" +
+      episodeSeason +
+      "E" +
+      episodeNumber;
+
+    //creating image container div//
+    let imageContainer = document.createElement("div");
+    imageContainer.classList.add("image-container");
+    movieCard.appendChild(imageContainer);
+
+    //creating episodeImage
+    let episodeImage = document.createElement("img");
+    imageContainer.appendChild(episodeImage);
+    episodeImage.src = episodeList[i].image.medium;
+
+    //creating summary div//
+    let summaryContainer = document.createElement("div");
+    summaryContainer.classList.add("summary-container");
+    movieCard.appendChild(summaryContainer);
+
+    //creating episodeSummary//
+    const episodeSummary = document.createElement("p");
+    summaryContainer.appendChild(episodeSummary);
+    summaryContainer.innerHTML = `${episodeList[i].summary}`;
+
+    ///creating div for level 300 and appended rootElm inside
+    let episodeDivContainer = document.createElement("div");
+    episodeDivContainer.classList.add("episode-container");
+    episodeDivContainer.id = "episode-container" + i;
+    rootElem.appendChild(episodeDivContainer);
+  }
+
+  //creating select bar for episodes for level 300
+
+  let selectElem = document.querySelector("#selector");
+  let optionElem = document.createElement("option");
+  optionElem.innerText = "Please Choose Your Episodes";
+  selectElem.appendChild(optionElem);
+
+  allEpisodes.forEach((ele) => {
+    let options = document.createElement("option");
+    options.value = ele.name;
+    options.innerText = `${ele.name} - S${ele.season
+      .toString()
+      .padStart(2, "0")}E${ele.number.toString().padStart(2, "0")}`;
+    selectElem.appendChild(options);
+  });
+
+  selectElem.addEventListener("change", function () {
+    let selectOption = selectElem.selectedIndex - 1;
+    let episodeDivContainer = document.getElementById(
+      "episode-container" + selectOption
+    );
+    episodeDivContainer.scrollIntoViewIfNeeded({ behavior: "smooth" });
+  });
+
+  //creating footer
+  let footerEle = document.getElementById("footer");
+  const footerLink = document.createElement("a");
+  footerLink.href = "https://www.tvmaze.com/";
+  const textNode = document.createTextNode(
+    "Copyright 2020. All data comes from: "
+  );
+  footerEle.appendChild(textNode);
+  footerLink.textContent = "www.tvmaze.com";
+  footerEle.appendChild(footerLink);
```

Comment:
This piece of code keeps appending footer on every change, see image below: 

![image](https://github.com/mabalal/tv-show-dom-project/assets/20670637/4a454905-1621-44a2-a133-9cc466d2572c)

We should try to avoid rendering or re-rendering stuff that's not changing. In this case, footer information should be extracted away from the function that renders episodes page, because footer doesn't change on episodes change :)

On a side note, React is going to great lengths to prevent this kind of behaviour with state and signals. If you are interested, I'd suggest reading about rendering in react - https://dev.to/mateo_garcia/understanding-rendering-in-react-i5i

Reply:
I extracted away the footer code out of function and now it seems displaying single footer. Thank you berkeli.